persist/fs/files.go

```diff
@@ -269,6 +306,24 @@ func findFiles(fileDir string, pattern string, fn toSortableFn) ([]string, error
 	return matched, nil
 }
 
+func findDirectories(osPath string) (map[string]string, error) {
```

Comment:
Perhaps `type fileBaseNamesToPaths map[string]string` to pass back a more readable type here?